visualize.py

```diff
@@ -3,36 +3,72 @@
 import csv
 
 
-def build_csv(output_file_name, data):
+def build_csv(output_file_name, results, headers, data_type):
+    multi_funcs = ["Mean", "Median", "Mode", "Standard deviation", "Variance"]
     if output_file_name.endswith('.csv') is False:
         output_file_name += ".csv"
     with open(output_file_name, 'w', newline='') as csv_file:
         write = csv.writer(csv_file)
-        write.writerow(['Function', 'Value'])
-        for function in data:
-            row = [function]
-            if type(data[function]) is list:
-                for result in data[function]:
-                    row.append(result)
-            else:
-                row.append(data[function])
-            write.writerow(row)
+        if data_type == "Interval":
+            write.writerow(['Function', 'Value'])
+            for function in results:
+                row = [function]
+                if type(results[function]) is list:
+                    if function in multi_funcs:
+                        for i in range(len(results[function])):
+                            if results[function][i] != results[function][-1]:
+                                row = [function + " " + headers[i+1]]
+                            else:
+                                row = [function + " difference"]
+                            row.append(results[function][i])
+                            write.writerow(row)
+                    else:
+                        for result in results[function]:
+                            row.append(result)
+                        write.writerow(row)
+                else:
+                    row.append(results[function])
+                    write.writerow(row)
+        # Data is ordinal
+        else:
+            write.writerow(['Function', 'Value'])
+            for function in results:
+                if type(results[function]) is list:
+                    for i in range(len(results[function])):
+                        row = [function + " #" + str(i+1), results[function][i]]
+                        write.writerow(row)
+        csv_file.close()
 
 
-def build_text(output_file_name, data):
+def build_text(output_file_name, results, headers, data_type):
     if output_file_name.endswith('.txt') is False:
         output_file_name += ".txt"
     with open(output_file_name, 'w', newline='') as csv_file:
         write = csv.writer(csv_file)
```

Comment:
Is build_text writing a csv file?

Reply:
I think it's writing a comma separated .txt file. It's definitely producing a .txt fie